### PR TITLE
Collections and workbooks UI improvements

### PR DESCRIPTION
**Pull request by imsitnikov**

### CODE REVIEW

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9273-workbooks-ui-improvements-707/component-collections-structure-707/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/units/collections/components/CollectionPage/hooks/useData.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        let getCollectionPromise: CancellablePromise<unknown> = new CancellablePromise(
+            Promise.resolve(),
+        );
+        let getCollectionBreadcrumbsPromise: CancellablePromise<unknown> = new CancellablePromise(
+            Promise.resolve(),
+        );
+
+        if (curCollectionId === null) {
+            dispatch(resetCollection());
+            dispatch(setCollectionBreadcrumbs([]));
```
Comment (stankis) [inline src/ui/units/collections/components/CollectionPage/hooks/useData.ts:87]: batch(() => { ... } ?

**Comment by Sergey-weber** [comment]
color for small collection icon should be same as big collection icon in dark theme
Screenshot 2024-03-19 at 16 36 58
Screenshot 2024-03-19 at 16 37 05

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

File: src/ui/units/collections/components/CollectionPage/hooks/useData.ts (hunk 1 above)
Comment (imsitnikov) [reply src/ui/units/collections/components/CollectionPage/hooks/useData.ts]: done

File: src/ui/components/AnimateBlock/AnimateBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export const AnimateBlock = ({className, delay = 0, offset = 100, children}: AnimateBlockProps) => {
+    const [playAnimation, setPlayAnimation] = React.useState<boolean>(false);
+
+    return (
+        <div className={b({animate: playAnimation}, className)}>
+            <Waypoint
+                // trigger animation if element is above screen
+                topOffset={'-100000%'}
+                bottomOffset={offset}
+                onEnter={async () => {
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply src/ui/components/AnimateBlock/AnimateBlock.tsx]: fixed

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

File: src/ui/components/CollectionsStructure/CreateEntryInWorkbookDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    open,
+    initialCollectionId = null,
+    entryType,
+    onApply,
+    onClose,
+}) => {
+    const history = useHistory();
+
+    const handleCreateEntry = React.useCallback(
+        async ({targetWorkbookId}: {targetWorkbookId: string | null}) => {
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply src/ui/components/CollectionsStructure/CreateEntryInWorkbookDialog.tsx]: yes, but this method must return a promise, add `return Promise.resolve()`.

File: src/ui/units/collections/components/CollectionContentTable/CollectionContentTable.tsx [hunk 4]
```diff
@@ -45,150 +71,145 @@ export const CollectionContentTable = React.memo<CollectionContentTableProps>(
...
+                                                : `${COLLECTIONS_PATH}/${item.collectionId}`
+                                        }
+                                        key={
+                                            'workbookId' in item
+                                                ? item.workbookId
+                                                : item.collectionId
+                                        }
                                         className={b('content-row')}
+                                        onClick={(e) => {
+                                            if (!e.metaKey) {
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply src/ui/units/collections/components/CollectionContentTable/CollectionContentTable.tsx]: I think it won't work on Windows, I've added check for `!e.ctrlKey` too. Thx for your attention.

**Review by stankis — APPROVED** [review]

File: src/ui/components/AnimateBlock/AnimateBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    const [playAnimation, setPlayAnimation] = React.useState<boolean>(false);
+
+    return (
+        <div className={b({animate: playAnimation}, className)}>
+            <Waypoint
+                // trigger animation if element is above screen
+                topOffset={'-100000%'}
+                bottomOffset={offset}
+                onEnter={() => {
+                    setTimeout(() => {
```
Comment (jhoncool) [inline src/ui/components/AnimateBlock/AnimateBlock.tsx:27]: I don't know why but I texted comment here. Anyway `setTimeout` is side effect. It should be inside the useEffect hook with clearTimeout

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

File: src/ui/components/AnimateBlock/AnimateBlock.tsx (hunk 5 above)
Comment (imsitnikov) [reply src/ui/components/AnimateBlock/AnimateBlock.tsx]: fixed – https://github.com/datalens-tech/datalens-ui/pull/707/commits/8c2d442732a89c508022554433918ea75e3fafd7

**Review by jhoncool — APPROVED** [review]

**Review by jhoncool — APPROVED** [review]

**Pull request merged**